### ISSUE TITLE
Curiosity

### ISSUE THREAD

**Issue by Marctraider** [loopy750/Windows-Input-Lag-Analysis#2]
Just curious, as you only use a 240fps camera, how many samples did you take?

Most tests that you do here should not show more than a few milliseconds of difference at best, you'd need a whole load of samples to actually get accurate results.

You'd also might want to retest disabledynamictick, as it is often touted as a magic fix to reducing input latency, it might actually be something that only 'reduces' deterministic latency and lowers jitter but not necessarily decreases input latency. It would in theory put more strain on certain systems which could cause increased latency.

Thats a theory, ofcourse.

**Comment by loopy750** [comment]
I tested each scenario with several samples and worked out an "average" as best I could, although in no cases did I ever actually see 4 frames of latency at 250Hz, so it's safe to assume the results are very "tight". A 1000 FPS camera would give finer results as I'd be able to better pinpoint the moment the mouse starts to move. Hence why results are approximate. The claim I wanted to prove/disprove (to myself) was different Windows versions have noticeable varying results in regards to input latency. In my opinion, even one frame of latency at 250Hz would be extremely hard to distinguish, but could easily be captured with a 240 FPS camera. However, to the best of my ability, as I was unable to record any difference at 240 FPS, I concluded that the results are "good enough" for me to not be concerned about using a specific Windows version over another.


In regards to disabledynamictick, I can agree with that. I am planning to further test it with a "clean" Windows 10 2004 install, as (from my brief testing) I am not convinced it helps in any meaningful way. I also haven't seen any negative effects either. It's very possible that with Windows 2004 the Default 'bcdedit' settings should be completely left as is in most if not all cases. (Of course, with millions of different hardware combinations, there may be no such thing as one-size-fits-all.)

**Comment by Marctraider** [comment]
Thanks for taking the time for this expansive explanation!

I too was caught in between Windows versions for a long while, but there's just too much 'good' in Windows 10 (When tweaked right) that outweight the potential 1ms lower latency (If any). I doubt there actually is.

If anything the difference (In full screen exlusive, both sides properly tweaked) is way less than 1ms :)
But you basically proved it yourself.

I'm doing some testing myself and trying to stay with 20 samples per test, bcdedit and 'game mode' are most interesting to me. Allegedly it reduces input latency by avg of 5ms (magically, because I dont really believe in fairy tales) but this was around Creators update, lots of unknowns, and MS basically said 180x+ no longer does any intrusive prioritizing.

So even if it somehow magically shaved 5ms off input (god knows how) where does that 5ms latency come from in an OS where things are processed in time frames of nanoseconds.

I actually expect zero difference on my extensively tweaked LTSC, but I would be surprised to see any difference what so ever with Game mode on.

Disabledynamictick in my book could potentially make a game feel more or less raw, more or less smooth, raw but more jitter vs less jitter and smoother feeling. This can also change things but not necessarily really reduce input latency by measurable amounts.

**Comment by loopy750** [comment]
Nice to hear. Any tests results of your own would be interesting to see and/or compare. There is a lot of (mis)information out there, but technology is cheap enough to do a lot of your own testing. Many phones these days have cameras with at least 240 FPS recording capabilities. And 1000 FPS are much cheaper than they used to be, although still quite expensive.

In every case where someone has claimed that X causes more/less latency and I've asked them how much latency exactly, they've never had an answer. So I decided to test things for myself. That's not to say that certain 'tweaks' don't benefit certain older hardware, but I'm not too interested in that just yet. I have a 10-year-old system with an older i7 from 2010, so that would be ideal for experimenting with how Windows 10 responds to older hardware with and without tweaks.

For now my researching is mostly based around modern hardware, such as i9 9900K, which is also why i recommend finding several sources and weighing up what sounds practical for your own personal hardware.

**Comment by Marctraider** [comment]
Yep, its pretty doable for anyone these days. 

Also got tired of all the misinformation and nothing ever gets backed up by numbers, and even if they are it doesn't show or reflect actual practical differences.

Stared to document myself, and preferably not even share it public as the so maniest 'Optimal tweaking guide number 494839'

I will share my data once I have some decently reliable numbers.

**Comment by loopy750** [comment]
Understood :)

If/when I do anything more, I'll probably just call it an explanation rather than a guide. I'll share what I did, what impact it has, if any, and people can decide whether or not it's also suitable for them.

**Comment by Marctraider** [comment]
disabledynamictick no, game mode off: 13.76ms
disabledynamictick yes, game mode off: 14.39ms
disabledynamictick no, game mode on: 14.29ms

Each took 20 samples, the results are all within margin of error really. Rebooted for game mode because unsure if it even applies directly. 

Basically if there is any change in disabledynamictick setting, it will be the aforementioned and not observable input latency.

Game mode doesn't seem to do anything on my already heavily tweaked and clean LTSC, so my conclusion on that is that it could only interfere with things.

I'll want to do more testing on various things but that will take some more time :P

**Comment by CiHel** [comment]
he say game mode reduces input frome https://youtu.be/mVNRNOcLUuA?t=601

1809 1903 1909 new game mode
https://www.reddit.com/r/pcgaming/comments/bzr6nx/does_windows_10_game_mode_still_hurt_performance/

"In the latest versions of Windows (1809 & 1903) Game Mode no longer does GPU/CPU prioritization. This prioritization was intended to give more resources to the running game at the cost of background processes, but we found it impacted some games and other scenarios. As an example streaming was impacted in apps like OBS because we were starving them of resources needed to encode.

In these later builds Game Mode is intended to help by removing distractions while playing. In particular it tries to stop Windows Update from updating drivers and stops it from notifying you of non-critical updates. It also causes the CPU to run at a minimum of 100% (on desktop, not laptops) to help reduce CPU fluctuations which may cause performance issues."

**Comment by Marctraider** [comment]
Yes, like I said "MS basically said 180x+ no longer does any intrusive prioritizing."

Didn't exactly elaborate on it, but it used to work differently. Still, on a clean and idle system (LTSC + Lots of crap disabled) I would still be skeptical about a free 5ms latency reduction, as prioritizing often means there is concurrency going on with a bloated system or lots of unnecessary crap running around in the background. (Basically any untouched clean install, including LTSC).

**Comment by Marctraider** [comment]
Low latency mode (from On) to Ultra: 13.55ms (Didn't expect any change with low gpu usage)
G-sync (No v-sync, middle of screen measured, 118fps (in-game): 17.14ms (Falls in line with blurbuster tests)

**Comment by Marctraider** [comment]
Didn't bother bothering you again, but on 1709 (with alleged beter game mode) i'm getting 11.43ms as result. Seems to be slightly snappier too.

20 samples is not enough to convince me yet, but that would shave off almost 3ms.

Hrm...

**Comment by loopy750** [comment]
Thanks for the testing.
Regarding 10MHz, Microsoft explains the changes made here (page 8): https://techcommunity.microsoft.com/gxcuf89792/attachments/gxcuf89792/NetworkingBlog/172/2/Evolution%20of%20Timekeeping%20in%20Windows.pdf
From the information I've read from people looking into the kernel, they've yet to find anything to suggest the changes are worse for gaming. And if anything they believe it could slightly improve performance. That's not to say this is 100% the case, but anyone suggesting the changes are bad for gaming have never really produced any specific evidence showing the timer is the cause of any said problems. So at this stage, seeing 10MHz as opposed to 3.xxMHz does not bother me in the slightest.

Possibly related to security patches and/or bios microcode updates (spectre/meltdown/zombieload)? 1709 might not contain security patches that are in later builds (that can potentially be disabled anyway)?
Also it could be related to hardware (including or not including security patches). "Back in the day", running the latest Windows was really only a viable option if you had the latest hardware. I tested an old dual core with Windows 10 once, which obviously is not recommended, and everything ran hot, and would not idle under 50%, whereas with Windows XP it idled at 0% and ran very smooth. Although a 9900K is not much newer than something a couple of years older, these security patches tend to affect them a lot more. So maybe a 9900K will run better on 1909 and something older will run better on 1709?

I guess if anything it raises more questions than answers. With my limited ability to test, I haven't seen anything like this. But even if it were the case of an almost 3ms improvement, it likely still wouldn't be enough to convince me to use an older build.

Basically although I think these tests are excellent (unsure of your current testing methods and hardware), the conclusion can be somewhat limited, unless you're able to test multiple CPU's and confirm it's the same results for all. And that also applies to anything I test. I can say with some confidence what does and doesn't work for a 9900K, but that won't necessarily be the case for older CPU's. I suppose that's why there aren't many people doing it...it can be rather costly.

**Comment by Marctraider** [comment]
Mhh I doubt security patches really significantly increase/decrease input latency by 3-5~ms unless FPS is affected. That'd be really weird. Not saying impossible though. 

I'd realistically have to compare LTSC, W7 and 1709 now. If 170x is the only branch (due to game mode) that shaves off a few ms more from baseline, it would be the only OS that has lower latency vs the rest. That would actually be something.

As for the timer, I haven't actually seen any performance decline myself, the timer could even be more accurate than normal (Altho doubtful this would really affect frametimes noticably enough)

The only thing we know is when testing with tool like TimerBench we get much less timer calls (I forgot exactly how much, but at least 2x less) with the new 10MHz synth timer.

**Comment by Marctraider** [comment]
Alright, last post from me and I'll close it.

Did testing on Windows 7, 10 (1709 and 1809). Assuming all is 'right', I basically get similar results to you when it comes to input latency on all of them. Around 7.5/8.5ms mouse to pixel latency (First touch to very first screen movement anywhere) or roughly 12~ms center of screen.

I can still not reproduce this magic game mode thing, and it would be impossible to substract this from my results you would get into territory which is impossible with my setup anyway.

Basically comes down to frametimes for me, any Windows 10 comes on top of that vs Windows 7, even on older games like CSGO.

**Comment by loopy750** [comment]
Thanks for sharing the results.

Speaking of frametimes, I've been using CapFrameX with Windows 10 2004. Of all the tweaks I tested from a fresh install, the only one that wasn't within margin of error is the spectre/meltdown registry entry that disables it.

That's not to say other tweaks don't do anything...it's more about understanding what each one does (and doesn't do). As some people have pointed out, doing a whole list of tweaks might show an improvement, but it's likely only 1 or 2 have made any significant difference, and other 100 have done basically nothing (or nothing noteworthy).
